pennylane/transforms/tape_expand.py

```diff
@@ -350,6 +353,33 @@ def custom_decomp_expand(self, circuit, max_expansion=decomp_depth):
     return custom_decomp_expand
 
 
+@transform
+def decomp_transform(tape: QuantumTape, custom_decomps: dict) -> (Sequence[QuantumTape], Callable):
```

Comment:
Any worries about making this user facing? With a bit of polish, we can have this transform version available and think about deprecation of  `set_decomposition` in 0.34.

Reply:
I also think _polish, publicize, deprecate_ makes sense.

Reply:
How about `custom_decomposition` as the name?